glossary.qmd

```diff
@@ -0,0 +1,353 @@
+
+#### Lossless compression
+
+A type of [compression](#compression) where the exact original values **can** be recovered after decompression. This means that the compression process does not lose any information. Lossless compression codecs tend to give larger file sizes than [lossy compression](#lossy-compression) codecs.
+
+Examples include [deflate](#deflate), [LZW](#lzw), [gzip](#gzip), and [ZSTD](#zstd).
+
+#### Lossy compression
+
+A type of [compression](#compression) where the exact original values **cannot** be recovered after decompression. This means that the compression process will lose information. Lossy compression codecs tend to give smaller file sizes than [lossless compression](#lossless-compression) codecs.
+
+Examples include [LERC](#lerc) and [JPEG](#jpeg).
+
+#### LZW
+
+A [lossless compression](#lossless-compression) codec for general use. It tends to be slightly slower than [deflate](#deflate).
+
+#### Mapbox Vector Tile
+
+A file format for tiled [vector data](#vector-data), usually used for visualization on web maps. [PMTiles](#pmtiles) is a [cloud-optimized](#cloud-optimized) [archive format](#archive-format) for storing millions of Mapbox Vector Tile files in an efficient manner, accessible via [HTTP range requests](#http-range-request).
+
+#### Metadata
+
+Information _about_ the actual data, saved as part of the file format. This allows for recreating the exact data that existed before saving and for [cloud-optimized](#cloud-optimized) data formats, usually stores the byte ranges of relevant data sections within the file, which allows for using [HTTP range requests](#http-range-request) for efficient [random access](#random-access) to that data section.
+
+#### Microsoft Azure
+
+[Cloud computing](#cloud) services offered by Microsoft.
+
+#### Multi-dimensional raster data
+
+A type of gridded [raster data](#raster-data) where multiple dimensions help conceptualize various attribtes. For example, a data value may exist for every longitude, latitude, time, and elevation, in which case the raster data would have four dimensions.
+
+#### Multithreading
+
+A manner of scaling computing, to allow more operations to happen at the same time.
+
+Think of a glass of water. Synchronous computing is akin to having one straw: when you've finished drinking all you wish to drink, you give the straw to your friend for them to drink. [Asynchronous](#asynchronous) computing is akin to sharing the straw between you and your friend. There's still only one straw, but you can hand off sips. Parallel computing (including multithreading and multiprocessing) is like having two straws, where both you and your friend can drink out of the glass at the same time.
+
+#### Numpy
+
+The foundational Python library for managing n-dimensional array data.
+
+#### Object storage (Cloud storage)
```

Comment:
Maybe this is the most practical explanation, but doesn't really explain key:value stores <- which implicates many things above behavior of such filesystems.

Reply:
I'm not sure what about key:value stores makes sense to include at this level of technicality.

Reply:
valid, also not all webservers are "cloud-storage" but for a user http range request is the important part.